session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -478,6 +483,25 @@ private void handleRedirection(String deviceId, EndPoint endpoint)
     }
   }
 
+  /**
+   * insert data in one row, if you want improve your performance, please use insertInBatch method
+   * or insertBatch method
+   *
+   * @param timeout  asynchronous call timeout in millisecond
+   * @param callback user provided callback if failed, set to null if user does not specify.
+   * @see Session#insertRecords(List, List, List, List, List)
+   * @see Session#insertTablet(Tablet)
+   */
+  public CompletableFuture<Integer> asyncInsertRecord(
+          String deviceId, long time, List<String> measurements, List<TSDataType> types,
+          List<Object> values, long timeout,
+          SixInputConsumer<String, Long, List<String>, List<TSDataType>, List<Object>, Throwable> callback) {
+    ExecutorService singleThreadPool = Executors.newFixedThreadPool(1);
```

Comment:
use a  shared thread pool for all Async interfaces. 
1. The size of the pool can be configured by users.
2. The thread pool can be provided by users themself. (e.g., session pool can provide a thread pool for all sessions. )
3. close the thread pool after session/sessionPool is closed.